src/signature.c

```diff
@@ -260,12 +283,14 @@ static bool get_pubkey(void)
 	cert = PEM_read_X509(fp_pubkey, NULL, NULL, NULL);
 	if (!cert) {
 		fprintf(stderr, "Failed PEM_read_X509() for %s\n", CERTNAME);
+		fp_pubkey = NULL;
```

Comment:
This was removed earlier to prevent a double free... https://github.com/clearlinux/swupd-client/pull/111/commits/e4f1e110cd0203984e39c708496ec045921e18df